File: backend/src/test/java/br/ufpr/sept/so2/modules/iam/domain/IdentificadorLoginTest.java
package br.ufpr.sept.so2.modules.iam.domain;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class IdentificadorLoginTest {

    @Test
    void normalizaEmailEGrr() {
        IdentificadorLogin email = IdentificadorLogin.tryParse("Aluno.Dev@UFPR.BR").orElseThrow();
        assertEquals(IdentificadorLogin.Tipo.EMAIL, email.getTipo());
        assertEquals("aluno.dev@ufpr.br", email.getValor());

        IdentificadorLogin grr = IdentificadorLogin.tryParse("grr20240001").orElseThrow();
        assertEquals(IdentificadorLogin.Tipo.GRR, grr.getTipo());
        assertEquals("GRR20240001", grr.getValor());

        IdentificadorLogin soDigitos = IdentificadorLogin.tryParse("20240001").orElseThrow();
        assertEquals("GRR20240001", soDigitos.getValor());
    }

    @Test
    void mascaraIdentificadorERejeitaLixo() {
        assertEquals("a***@ufpr.br", IdentificadorLogin.tryParse("aluno.dev@ufpr.br").orElseThrow().mascarado());
        assertEquals("GRR****0001", IdentificadorLogin.tryParse("GRR20240001").orElseThrow().mascarado());
        assertTrue(IdentificadorLogin.tryParse("nao-e-identificador").isEmpty());
        assertTrue(IdentificadorLogin.tryParse("").isEmpty());
    }
}

File: backend/src/test/kotlin/br/ufpr/sept/so2/modules/iam/domain/IdentificadorLoginTest.kt
package br.ufpr.sept.so2.modules.iam.domain

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class IdentificadorLoginTest : StringSpec({
    "normaliza e-mail e GRR" {
        val email = IdentificadorLogin.tryParse("Aluno.Dev@UFPR.BR")!!
        email.tipo shouldBe IdentificadorLogin.Tipo.EMAIL
        email.valor shouldBe "aluno.dev@ufpr.br"

        val grr = IdentificadorLogin.tryParse("grr20240001")!!
        grr.tipo shouldBe IdentificadorLogin.Tipo.GRR
        grr.valor shouldBe "GRR20240001"

        val soDigitos = IdentificadorLogin.tryParse("20240001")!!
        soDigitos.valor shouldBe "GRR20240001"
    }

    "mascara identificador e rejeita lixo" {
        IdentificadorLogin.tryParse("aluno.dev@ufpr.br")!!.mascarado() shouldBe "a***@ufpr.br"
        IdentificadorLogin.tryParse("GRR20240001")!!.mascarado() shouldBe "GRR****0001"
        IdentificadorLogin.tryParse("nao-e-identificador") shouldBe null
        IdentificadorLogin.tryParse("") shouldBe null
    }
})
